registry/remote/auth/client.go

```diff
@@ -223,17 +228,25 @@ func (c *Client) Do(originalReq *http.Request) (*http.Response, error) {
 	case SchemeBearer:
 		resp.Body.Close()
 
-		// merge hinted scopes with challenged scopes
-		scopes := GetScopes(ctx)
-		if scope := params["scope"]; scope != "" {
-			scopes = append(scopes, strings.Split(scope, " ")...)
+		scopes := GetScopesPerHost(ctx, host)
+		cleanScopeLen := len(scopes)
+		if moreScopes := GetScopes(ctx); len(moreScopes) > 0 {
+			// merge per-host scopes with generic scopes
+			scopes = append(scopes, moreScopes...)
+		}
```

Comment:
The code related to `GetScopes()` should be done inside of `GetScopesPerHost()`.

Reply:
But users may not want to get a mixed result of scopes and per-host scopes when invoking ` GetScopesPerHost()`.

Reply:
Added a private helper function for this.